apicast/src/configuration.lua

```diff
@@ -66,16 +66,23 @@ local function check_rule(req, rule, usage_t, matched_rules, params)
 end
 
 local function get_auth_params(method)
-  local params
+  local params = ngx.req.get_uri_args()
 
   if method == "GET" then
-    params = ngx.req.get_uri_args()
+    return params
   else
     ngx.req.read_body()
-    params = ngx.req.get_post_args()
-  end
+    local body_params = ngx.req.get_post_args()
+
+    -- Adds to body_params URI params that are not included in the body. Doing
+    -- the reverse would be more expensive, because in general, we expect the
+    -- size of body_params to be larger than the size of params.
+    for k, v in pairs(params) do
```

Comment:
This is not JITable. http://wiki.luajit.org/NYI#libraries_base-library

What about using metatable?

```lua
setmetatable(body_params, { __index = params })
```

That will just allocate one extra table, but no need for the iteration.

Reply:
:+1: